src/containers/clock.jsx

```diff
@@ -0,0 +1,164 @@
+import sortByTime from '../utils/sort';
+import WatchList from '../components/watch-list';
+
+
+const styles = theme => ({
+  root: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    minHeight: '100vh',
+    backgroundColor: '#000',
+    textAlign: 'center',
+    flexDirection: 'column',
+  },
+});
+
+const StopwatchControls = ({ onToggle, running}) =>
+  <Fab onClick={onToggle}>
+    {running ? 'Stop' : 'Start'}
+  </Fab>
+
+const bestTime = (attempts) => {
+  if (attempts.length === 0){
+    return '00:00:00';
+  }
+  const best = attempts.sort(sortByTime)[0];
+  return Stopwatch({ ...best, format: 'home' })
+}
+
+const averageTime = (attempts) => {
+  if (attempts.length === 0){
+    return '00:00:00';
+  }
+  const average = attempts.reduce((acc, current) => acc + current.time, 0) / attempts.length;
+  return Stopwatch({ time: average, format: 'home' });
+}
+
+
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      time: 0,
+      running: false,
+      date: '',
+    };
+    this.toggle = this.toggle.bind(this);
+    this.tick = this.tick.bind(this);
+  }
+  toggle() {
+    if (!this.state.running) {
+      this.startTime = +new Date();
+      this.tick();
+    } else {
+      const dataLocal = JSON.parse(localStorage.getItem('attempts'));
+      if (dataLocal){
+        dataLocal.push({ time: this.state.time, date: new Date() })
+        localStorage.setItem('attempts', JSON.stringify(dataLocal));
+      } else{
+        localStorage.setItem('attempts', JSON.stringify([{ time: this.state.time, date: new Date() }]))
+      }
+      this.update(false);
+      this.setState({ time: 0 });
+      clearTimeout(this.timeout);
+    }
+  }
+  tick() {
+    this.update(true);
+    this.timeout = setTimeout(this.tick);
+  }
+  update(running) {
+    this.setState({
+      time: +new Date() - this.startTime,
+      running
+    });
+  }
+  render() {
+    const { classes } = this.props;
+    const attempts = JSON.parse(localStorage.getItem('attempts')) || [];
+    return (
+      <div className={classes.root}>
+        <div className="stopwatch">
+          <h1 style={{ color: '#fff' }}>Stopwatch</h1>
+          <section>
+            <div style={{ color: '#fff'}}>
+              <Stopwatch time={this.state.time} format='time' />
+            </div>
+            <StopwatchControls onToggle={this.toggle} running={this.state.running} />
+          </section>
+          <section>
+            {attempts.length > 0 && (
+              <WatchList attempts={attempts} format='home' />
+            )}
+          </section>
+          <section style={{ display: 'flex', justifyContent: 'space-between', margin: '16px 0' }}>
+            <Link to={`/`}>
+              <Chip
+                icon={<DeleteIcon />}
+                label="Reset all"
+                // clickable
+                onClick={() => localStorage.removeItem('attempts')}
```

Comment:
I'd recommend to put this callback into a function 

could be `clearAll` or `resetAll`